roles/openshift_prometheus/files/node-exporter-template.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: prometheus-node-exporter
+  annotations:
+    openshift.io/display-name: Prometheus Node Exporter
+    description: Prometheus exporter for node host metrics
+    iconClass: fa fa-cogs
+    tags: monitoring,prometheus
+    openshift.io/support-url: https://access.redhat.com
+    openshift.io/provider-display-name: Red Hat, Inc.
+parameters:
+- name: IMAGE
+  value: openshift/prometheus-node-exporter:v0.15.2
+- name: MEMORY_REQUESTS
+  value: 30Mi
+- name: CPU_REQUESTS
+  value: 100m
+- name: MEMORY_LIMITS
+  value: 50Mi
+- name: CPU_LIMITS
+  value: 200m
+objects:
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    name: prometheus-node-exporter
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
+      prometheus.io/scrape: "true"
+    labels:
+      app: prometheus-node-exporter
+    name: prometheus-node-exporter
+  spec:
+    clusterIP: None
+    ports:
+    - name: scrape
+      port: 9100
+      protocol: TCP
+      targetPort: 9100
+    selector:
+      app: prometheus-node-exporter
+- apiVersion: extensions/v1beta1
+  kind: DaemonSet
+  metadata:
+    name: prometheus-node-exporter
+    labels:
+      app: prometheus-node-exporter
+      role: monitoring
+  spec:
+    updateStrategy:
+      type: RollingUpdate
+    template:
+      metadata:
+        labels:
+          app: prometheus-node-exporter
+          role: monitoring
+        name: prometheus-exporter
+      spec:
+        serviceAccountName: prometheus-node-exporter
+        hostNetwork: true
+        hostPID: true
+        containers:
+        - image: ${IMAGE}
+          args:
+          - "--collector.procfs=/host/proc"
+          - "--collector.sysfs=/host/sys"
```

Comment:
@spadgett there are lots of CLI opts for this app but I can't figure out how to expose this. Is there a way to use yedit so an arbitrary list of args could be passed in?

Reply:
You could make `args` a parameter in the template like `${{ARGS}}` or you could make easy arg a separate template param.

@deads2k Any strong opinion?

Reply:
I'm not sure what you mean. I tried this but kube doesn't accept it as a valid config:

```
args: ["--collector.procfs=/host/proc", "--collector.sysfs=/host/sys"]
```

I'd be happy to follow an existing example if we can find one.

Reply:
I mean you could use the `${{ }}` template parameter syntax to pass an array instead of a string if you wanted.

https://docs.openshift.org/latest/dev_guide/templates.html#writing-parameters

> When using the ${{PARAMETER_NAME}} syntax only a single parameter reference is allowed and leading/trailing characters are not permitted. The resulting value will be unquoted unless, after substitution is performed, the result is not a valid json object. If the result is not a valid json value, the resulting value will be quoted and treated as a standard string.

Reply:
I assume that works for arrays, but maybe only JSON objects. @bparees ?

Reply:
should work for anything.  it just results in a direct replace of the value, so it's up to you to ensure the end result is valid json syntax after the substitution.